tests/test_flink_tools.py

```diff
@@ -319,3 +319,529 @@ def test_non_bool_spot_value(self):
             branch_dict=None,
         )
         assert flink_deployment_config.get_pool() == "flink-spot"
+
+
+# Tests for newly added formatting and data collection functions
+
+
+class TestCollectFlinkJobDetails:
+    """Tests for collect_flink_job_details function."""
+
+    def test_collect_with_no_overview(self):
+        """Test collecting job details when cluster is not running."""
+        status = {
+            "state": "stopped",
+            "pod_status": [
+                {"phase": "Running"},
+                {"phase": "Running"},
+            ],
+        }
+        overview = None
+        jobs = []
+
+        result = flink_tools.collect_flink_job_details(status, overview, jobs)
+
+        assert result["state"] == "stopped"
+        assert result["pod_counts"]["running"] == 2
+        assert result["pod_counts"]["evicted"] == 0
+        assert result["pod_counts"]["other"] == 0
+        assert result["pod_counts"]["total"] == 2
+        assert result["job_counts"] is None
+        assert result["taskmanagers"] is None
+        assert result["slots_available"] is None
+        assert result["slots_total"] is None
+        assert result["jobs"] == []
+
+    def test_collect_with_evicted_pods(self):
+        """Test collecting pod details with evicted pods."""
+        status = {
+            "state": "running",
+            "pod_status": [
+                {"phase": "Running"},
+                {"phase": "Failed", "reason": "Evicted"},
+                {"phase": "Failed", "reason": "Evicted"},
+                {"phase": "Pending"},
+            ],
+        }
+        overview = mock.Mock(autospec=True)
+        overview.jobs_running = 1
+        overview.jobs_finished = 2
+        overview.jobs_failed = 0
+        overview.jobs_cancelled = 1
+        overview.taskmanagers = 3
+        overview.slots_available = 5
+        overview.slots_total = 15
```

Comment:
`overview` is a typeddict, no? do we need a mock here? can't we just construct the expected dict?

Reply:
(same for other tests below)